src/Arcus.WebApi.Security.Authentication/SharedAccessKey/HeaderSharedAccessKeyAuthenticationFilter.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Arcus.Security.Secrets.Core.Exceptions;
+using Arcus.Security.Secrets.Core.Interfaces;
+using GuardNet;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Primitives;
+
+namespace Arcus.WebApi.Security.Authentication.SharedAccessKey
+{
+    /// <summary>
+    /// Authentication filter to secure HTTP requests with shared access keys.
+    /// </summary>
+    /// <remarks>
+    ///     Please provide an <see cref="ISecretProvider"/> implementation in the configured services of the request.
+    /// </remarks>
+    public class HeaderSharedAccessKeyAuthenticationFilter : SharedAccessKeyAuthenticationFilter
```

Comment:
A lot of duplication is happening because we have two types (filter & attribute) for each config option, + an abstracted type that don't has a lot of functionality to be an abstracted class.

Reply:
We also have an `CertificateAuthentication` config option that has several config options (what of the certificate value to validate, and where the expected certificate value is located); but the attribute of that authentication mechanism is very straightforward so we can do the config somewhere else.

I would also suggest to have some kind of approach that would be in line what that. If you consider that we introduce yet another place to find the shared access key, we would need a lot of types to make that happen.

Reply:
The `CertificateAuthentication` has a `...Validator` and `...Config` combination to make this happen. And an enum is used to indicate to location where to look.

Reply:
It would decrease the complexity of the attribute, move the config to the `Startup` and make the maintainability better for us.

Reply:
But let's hear what @tomkerkhove thinks.

Reply:
@stijnmoreels first thanks for looking at this. I had not looked at the `CertificateAuthentication` implementation yet, so I will have a look at that. Reason I created the abstract class and the types per configuration is that I did not like to have `if..else` constructs to determine the type of authentication to perform.
@tomkerkhove set up a Teams meeting this afternoon to discuss some changes.